core/utils/bit_packing.cpp

```diff
@@ -252,9 +250,9 @@ template<int N>
 void fastpack(const uint64_t* IRS_RESTRICT in,
               uint64_t* IRS_RESTRICT out) noexcept {
   // 64 == sizeof(uint64_t) * 8
-  static_assert(N > 0 && N <= 64, "N <= 0 || N > 64");
-  // ensure all computations are constexr, i.e. no conditional jumps, no loops,
-  // no variable increment/decrement
+  static_assert(0 < N && N < 64);
+  // ensure all computations are constexpr,
+  // i.e. no conditional jumps, no loops, no variable increment/decrement
   *out |= ((*in) % (1ULL << N)) << (N * 0) % 64;
```

Comment:
What's the purpose of the bitshift by `(N * 0)` (i.e. 0) bits here?

Reply:
I think it's just same way to write different iterations of this functions

Reply:
Maybe, but bitshifting something by 0 still seems kind of useless. Why not just remove the `<< (N * 0)` part here?